processor/abci.go

```diff
@@ -878,6 +904,14 @@ func (app *App) RequireValidatorMasterPubKey(ctx context.Context, tx abci.Tx) er
 	return nil
 }
 
+func (app *App) DeliverProtocolUpgradeCommand(ctx context.Context, tx abci.Tx) error {
+	pu := &commandspb.ProtocolUpgradeProposal{}
+	if err := tx.Unmarshal(pu); err != nil {
+		return err
+	}
+	return app.protocolUpgradeService.UpgradeProposal(ctx, tx.PubKeyHex(), pu.Proposal.UpgradeBlockHeight, pu.Proposal.ReleaseTag)
```

Comment:
I'm glad to see the update height is not longer compiled in now, but who is now in charge of choosing it?

Reply:
Validators should discuss it together and fix a block height together.